src/shogun/base/unique.h

```diff
@@ -0,0 +1,44 @@
+#ifndef __SG_UNIQUE_H__
+#define __SG_UNIQUE_H__
+
+#ifdef HAVE_CXX11
+
+#include <memory>
+
+namespace shogun
```

Comment:
Hmm probably I should do some magic here to push all the std stuff to .cpp.

We can't ignore it in SWIG because it would be needed through concrete classes anyway